### PR TITLE
Switch to GLOBAL_TIMEOUT

### LINKED ISSUE #1845: [CI/CD] Add a global timeout for the CI pipeline runs
Recently, we had some hanging stages for Bors that blocked the CI. This needed a manual abort in order to unblock the CI.

We should have a global timeout (let's say 4 hours) after which we should terminate every hanging stage (if present).

### PR DESCRIPTION
**Pull request by vlaza**
Fixes #1845 by integrating GLOBAL_TIMEOUT into Jenkinsfiles

### CODE REVIEW

**Comment by vlaza** [comment]
bors try

**Comment by ionutbalutoiu** [comment]
bors try

**Comment by vlaza** [comment]
bors try

File: .jenkins/Jenkinsfile [hunk 1]
```diff
@@ -1,32 +1,37 @@
 @Library("OpenEnclaveCommon") _
 oe = new jenkins.common.Openenclave()
+GLOBAL_TIMEOUT = 240
```
Comment (aserdean) [inline .jenkins/Jenkinsfile]: Just one small nit on this one. TBH I have no idea what 240 is measuring. Mind adding a comment to inform users if this is in seconds/minutes?

File: .jenkins/Nightly_packages_testing.Jenkinsfile [hunk 2]
```diff
@@ -1,17 +1,20 @@
 @Library("OpenEnclaveCommon") _
 oe = new jenkins.common.Openenclave()
+GLOBAL_TIMEOUT = 240
```
Comment (aserdean) [inline .jenkins/Nightly_packages_testing.Jenkinsfile]: Same as above.

File: .jenkins/Packaging.Jenkinsfile [hunk 3]
```diff
@@ -1,31 +1,33 @@
 @Library("OpenEnclaveCommon") _
 oe = new jenkins.common.Openenclave()
+GLOBAL_TIMEOUT = 240
```
Comment (aserdean) [inline .jenkins/Packaging.Jenkinsfile]: Same as above.

File: .jenkins/build_docker_images.Jenkinsfile [hunk 4]
```diff
@@ -1,66 +1,70 @@
 @Library("OpenEnclaveCommon") _
 oe = new jenkins.common.Openenclave()
+GLOBAL_TIMEOUT = 240
```
Comment (aserdean) [inline .jenkins/build_docker_images.Jenkinsfile]: Same as above.

File: .jenkins/build_vhd.Jenkinsfile [hunk 5]
```diff
@@ -1,25 +1,28 @@
 @Library("OpenEnclaveCommon") _
 oe = new jenkins.common.Openenclave()
+GLOBAL_TIMEOUT = 240
```
Comment (aserdean) [inline .jenkins/build_vhd.Jenkinsfile]: Same as above.

File: .jenkins/libcxx_tests.Jenkinsfile [hunk 6]
```diff
@@ -3,6 +3,7 @@ import hudson.model.*
 
 @Library("OpenEnclaveCommon") _
 oe = new jenkins.common.Openenclave()
+GLOBAL_TIMEOUT = 240
```
Comment (aserdean) [inline .jenkins/libcxx_tests.Jenkinsfile]: Same as above.

File: .jenkins/Jenkinsfile (hunk 1 above)
Comment (vlaza) [reply .jenkins/Jenkinsfile]: I have addressed with a comment in all Jenkinsfiles

**Comment by vlaza** [comment]
bors try

File: .jenkins/Jenkinsfile (hunk 1 above)
Comment (ionutbalutoiu) [reply .jenkins/Jenkinsfile]: Good idea! @vlaza please specify that it's minutes everywhere.

**Pull request merged**